src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/ResourceActions/ResourceActions.ts

```diff
@@ -70,12 +77,30 @@ export var resourceActionsDirective = (
             cancel: "=?",
             edit: "=?",
             moderate: "=?",
+            modals: "=?"
         },
         templateUrl: adhConfig.pkg_path + pkgLocation + "/ResourceActions.html",
         link: (scope, element) => {
             var path = scope.parentPath ? AdhUtil.parentPath(scope.resourcePath) : scope.resourcePath;
             scope.modals = new Modals($timeout);
             adhPermissions.bindScope(scope, path, "options");
+
+            var badgeAssignmentPoolPath;
+            scope.$watch("resourcePath", (resourcePath) => {
+                if (resourcePath) {
+                    adhHttp.get(resourcePath).then((badgeable) => {
+                        badgeAssignmentPoolPath = badgeable.data[SIBadgeable.nick].post_pool;
+                    });
+                }
+            });
+            adhPermissions.bindScope(scope, () => badgeAssignmentPoolPath, "badgeAssignmentPoolOptions");
+            var params = {
+                depth: 4,
+                content_type: SIBadge.nick
+            };
+            adhHttp.get(scope.processUrl, params).then((response) => {
+                scope.badgesExist = response.data[SIPool.nick].count > 0;
+            });
```

Comment:
> This (expensive) code is executed even if scope.assignBadges is falsy. I would prefer if this was inside of adhBadgesActionDirective.

I added a check
```
if (scope.assignBadges) { ... }
```
around this logic. The parameters are needed in `ResourceActions.html`.